implementations/rust/ockam/ockam_transport_tcp/src/router.rs

```diff
@@ -136,3 +136,21 @@ impl TcpRouter {
         Ok(TcpRouterHandle { ctx, addr })
     }
 }
+
+/// Format an address of for a peer in this address type
+pub trait PeerAddressFormatter {
+    fn peer_addr<S: ToString>(_: S) -> Address;
+}
+
+/// Address format for TCP peers
+impl PeerAddressFormatter for TcpRouter {
+    fn peer_addr<S: ToString>(s: S) -> Address {
+        format!("{}#{}", ADDRESS_TYPE_TCP, s.to_string()).into()
+    }
+}
```

Comment:
I like the idea of having a common interface to construct addresses of a certain type, but I'm not so sure about associating the function with the `TcpRouter` (or actually any particular type).

It might not be immediately obvious to a user why they should expect this function on the Router, vs (for example) the connection worker handle.

I think we might be able to get away with just having a convention that transport crates expose a function in their root called `address` (and enforce this with some tests maybe), so that users of the crate can simply call `tcp::address("127.0.0.1:1234")` (provided they `use ockam_transport_tcp as tcp` first).

Reply:
That sounds like a good idea.